ui/dashboard.ts

```diff
@@ -1,25 +1,24 @@
-require("./zerorpc_client.js")();
-var settings = require("./settings.js")();
-require("./user_settings.js")();
-require("./monitor.js")();
-require("./utils.js")();
-require("./exchange.js")();
-require("./balances_table.js")();
-require("./navigation.js")();
-require("./userunlock.js")();
-require("./topmenu.js")();
+import { assert_exchange_exists, Currency, format_currency_value, settings } from './settings';
+import { create_or_reload_exchange } from './exchange';
+import { change_location, determine_location } from './navigation';
+import client from './zerorpc_client';
+import { get_total_asssets_value, iterate_saved_balances, setup_log_watcher, showError, setup_client_auditor } from './utils';
+import { set_ui_main_currency } from './topmenu';
+import { prompt_sign_in } from './userunlock';
+import { total_table_add_balances, total_table_recreate } from './balances_table';
+import { monitor_add_callback } from './monitor';
 
 function add_exchange_on_click() {
-    $('.panel a').click(function(event) {
+    $('.panel a').click(function(event: any) {
         event.preventDefault();
-        var target_location = determine_location(this.href);
+        let target_location = determine_location(this.getAttribute('href'));
         if (target_location.startsWith('exchange_')) {
             let exchange_name = target_location.substring(9);
             assert_exchange_exists(exchange_name);
-            console.log("Going to exchange " + exchange_name);
+            console.log('Going to exchange ' + exchange_name);
             create_or_reload_exchange(exchange_name);
         } else {
-            throw "Invalid link location " + target_location;
+            throw new Error('Invalid link location ' + target_location);
```

Comment:
What's the difference here? What does `new Error('XXX')` do compared to a simple throw?